OpenSim/Simulation/GeodesicWrap/GeodesicPath.h

```diff
@@ -0,0 +1,85 @@
+#ifndef OPENSIM_GEODESIC_PATH_H_
+#define OPENSIM_GEODESIC_PATH_H_
+
+// INCLUDE
+
+#include "OpenSim/Simulation/GeodesicWrap/GeodesicCurveState.h"
+#include "GeodesicWrapObject.h"
+#include "GeodesicPathSolver.h"
+
+namespace OpenSim {
+
+// Notes:
+// - rename Object to Surface?
+// - GeodesicWrapObject holds pointer to frame, how?
+// - ImplicitSurface holds the local state, how can it cache it?
+// - what is the workflow?
+//     - create surface component, add to model, connect to frame.
+//     - create wrapping path, add to model, connect each surface.
+//     - what are those connections? Sockets? Parameters? shared-pointers?
+
+class AbstractGeometryPath;
+
+// Concrete implementation of the AbstractGeometryPath.
+//
+// Contains the solver for computing the entire wrapping path.
+//
+// Comparable to GeometryPath?
+//
+// Does the following:
+// - Caches the result from the solver.
+// - Holds set of WrapOstacles
+// - Implements AbstractGeometryPath
+class OSIMSIMULATION_API GeodesicPath : public AbstractGeometryPath {
+OpenSim_DECLARE_CONCRETE_OBJECT(GeodesicPath, ModelComponent);
+public:
+    //==============================================================================
+    // PROPERTIES
+    //==============================================================================
+    OpenSim_DECLARE_PROPERTY(some_property, std::string,
+        "some property");
+
+    //--------------------------------------------------------------------------
+    // CONSTRUCTION
+    //--------------------------------------------------------------------------
+public:
+    GeodesicPath() = default;
+    ~GeodesicPath() = default;
+
+//=============================================================================
+// METHODS
+//=============================================================================
+
+    //--------------------------------------------------------------------------
+    // ABSTRACTGEOMETRYPATH IMPLEMENTATION
+    //--------------------------------------------------------------------------
+    double getLength(const SimTK::State& s) const override;
+    double getLengtheningSpeed(const SimTK::State& s) const override;
+    void addInEquivalentForces(const SimTK::State& state,
+    const double& tension,
+    SimTK::Vector_<SimTK::SpatialVec>& bodyForces,
+    SimTK::Vector& mobilityForces) const override;
+    double computeMomentArm(const SimTK::State& s,
+    const Coordinate& aCoord) const override;
+    bool isVisualPath() const override;
+
+private:
+
+    // Path start and end points.
+    SimTK::Vec3 _pStart;
+    SimTK::Vec3 _pEnd;
+
+    // List of wrap objects.
+    std::vector<GeodesicWrapObject> _wrapObstacles;
```

Comment:
How would a `GeodesicWrapObject` get added to the path? I think you'd need to pass:
- The `Surface` model component to create the wrap object against (e.g., `Sphere`).
- Which form to use (e.g., "implicit" for `ImplicitSphere`).
- An initial `GeodesicCurveState`  (maybe this is optional and we have a default state).

Reply:
(I vote for renaming to `GeodesicPathSegment` :)  )

I thought that, similar as it works now, that you create a `Path` component, and you connect it to the `Surface` components. The `Surface` is able to construct a `GeodesicPathSegment/GeodesicWrapObject` by giving it the correct parameters, and handing it the body index to the underlying transform.
The surface might create an implicit or parametric form, but we cannot really know at this level, because we only see the `GeodesicWrapObject` which is the base of both.
The initial state I have working that you construct during the first wrapping attempt. So no need to do anything during the obj construction I think